src/js/recommendation-ui.js

```
@@ -0,0 +1,227 @@
+        if (typeof toggleCompare === 'function') {
+          toggleCompare(e, name);
+          // Optimistically toggle visual state
+          const isCurrentlyActive = compareBtn.classList.contains('text-primary');
+          if (isCurrentlyActive) {
+             compareBtn.classList.remove('text-primary');
+             compareBtn.classList.add('text-zinc-400');
+             compareBtn.innerHTML = '<span class="material-symbols-outlined text-sm">compare_arrows</span> Compare';
+             card.classList.remove('ring-2', 'ring-primary/30');
+          } else {
+             // check if limit met safely
+             const curCompareList = typeof window.compareList !== 'undefined' ? window.compareList : (typeof compareList !== 'undefined' ? compareList : []);
```

ESLint

[error] 'compareList' is not defined.

(no-undef)